common/src/main/java/com/spredfast/kafka/connect/s3/S3.java

```diff
@@ -1,32 +1,31 @@
 package com.spredfast.kafka.connect.s3;
 
-import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
-import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import java.net.URI;
 import java.util.Map;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
 
 public class S3 {
 
-  public static AmazonS3 s3client(Map<String, String> config) {
+  public static S3Client s3client(Map<String, String> config) {
 
-    AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
+    S3ClientBuilder builder = S3Client.builder();
 
     String s3Region = config.get("s3.region");
     String s3Endpoint = config.get("s3.endpoint");
 
     // If worker config sets explicit endpoint override (e.g. for testing) use that
     if (s3Endpoint != null) {
-      if (s3Region == null) {
-        throw new IllegalArgumentException("s3.region must be set if s3.endpoint is set");
-      }
+      builder = builder.endpointOverride(URI.create(s3Endpoint));
+    }
```

Comment:
The previous code had a guard in place to prevent the endpoint from being specified without the region. This works fine for `localstack` as the region is not required but in production this configuration could produce an error.

Reply:
I thought the guard was in place just because the `EndpointConfiguration` constructor required it. But I'll bring it back so that we don't change the original behavior.

Reply:
Done.